From ebccc8e4509d3197f56316ba0bfff9d28574bb57 Mon Sep 17 00:00:00 2001
From: Chen Liangjun <b36089@freescale.com>
Date: Thu, 29 Dec 2011 14:27:27 +0800
Subject: [PATCH 263/442] ENGR00171038 ASRC:bug fix for convertion to 16k

16k output samplerate is not support in our driver and the call of
config a pair IOCTL function would return -1 to program. The test
program should stop the further ASRC operation when -1 is
received.however, because the missing of brackets, the err is given a
wrong value 1(it should be -1).so it will hold the further operation of
ASRC and access some data not defined or so. thus system hang.

Add the missing brackets.

Signed-off-by: Chen Liangjun <b36089@freescale.com>
---
 test/mxc_asrc_test/mxc_asrc_test.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/mxc_asrc_test/mxc_asrc_test.c b/test/mxc_asrc_test/mxc_asrc_test.c
index e34fe95..ceda7e5 100644
--- a/test/mxc_asrc_test/mxc_asrc_test.c
+++ b/test/mxc_asrc_test/mxc_asrc_test.c
@@ -133,7 +133,7 @@ int configure_asrc_channel(int fd_asrc, struct audio_info_s *info)
 	config.inclk = inclk;
 	config.outclk = outclk;
 	pair_index = req.index;
-	if (err = ioctl(fd_asrc, ASRC_CONFIG_PAIR, &config) < 0)
+	if ((err = ioctl(fd_asrc, ASRC_CONFIG_PAIR, &config)) < 0)
 		return err;
 
 	for (i = 0; i < config.buffer_num; i++) {
-- 
1.8.0

